File: RecoverySense_Starter_R/apps/wear_android/app/src/main/java/edu/uci/recoverysense/wear/SensorReading.kt
package edu.uci.recoverysense.wear

data class SensorReading(
    val timestamp: Long = System.currentTimeMillis(),
    val heartRate: Float = 0f,
    val accelX: Float = 0f,
    val accelY: Float = 0f,
    val accelZ: Float = 0f,
) {
    fun toJson(): String {
        return """
            {
              "timestamp": "$timestamp",
              "heartRate": $heartRate,
              "accelX": $accelX,
              "accelY": $accelY,
              "accelZ": $accelZ
            }
        """.trimIndent()
    }
}

File: RecoverySense_Starter_R/apps/phone_flutter/lib/src/features/sensors/sensor_reading.dart
class SensorReading {
  final DateTime timestamp;
  final double? heartRate;
  final double? accelX;
  final double? accelY;
  final double? accelZ;

  const SensorReading({
    required this.timestamp,
    this.heartRate,
    this.accelX,
    this.accelY,
    this.accelZ,
  });

  Map<String, dynamic> toJson() => {
        'timestamp': timestamp.toIso8601String(),
        'heartRate': heartRate,
        'accelX': accelX,
        'accelY': accelY,
        'accelZ': accelZ,
      };
}
